### DISCUSSION TITLE
Sample App and Sample Libraries Repos

### DISCUSSION THREAD

**Discussion by cipolleschi** [reactwg/react-native-new-architecture#26] [Documentation]
Hi everyone, 
This thread’s goal is to officially present a couple of repositories we created to support the community while migrating from the old architecture (React Native’s versions < 0.68.0) to the New Architecture (React Native’s versions >= 0.68.0).

The two repositories are:
[RNNewArchitectureApp](https://github.com/react-native-community/RNNewArchitectureApp)
[RNNewArchitectureLibraries](https://github.com/react-native-community/RNNewArchitectureLibraries)

The first repository contains runs of migrating an app from the Old Architecture to the New Architecture, following the steps shown in the [Guide](https://reactnative.dev/docs/next/new-architecture-intro). Every run has its own branch and every commit in the run should build and run properly.

The second repository is focused on Libraries, with a particular care on backward compatibility. The runs guide you in the creation of a TurboModule or a Fabric Component, ensuring that they can work on both the Old and the New Architecture.

These repositories are thought of as supporting material for the official documentation. Check every branch: each branch represents a different run with different features.

Any feedback on them is greatly appreciated.

**Comment by gorhom** [comment]
@cipolleschi thanks for sharing these repos, they are really helpfully!

I was wondering if it possible to create another branch example for libraries but with shared C++ logic between iOS and Android ?

**Reply by cipolleschi** [reply]
Hi @gorhom, thanks for your question.

That's a good idea! At the moment we are a bit busy, but we will likely produce something on those lines at some point in the future.

**Reply by alon7** [reply]
Hi @cipolleschi we already implemented it in https://github.com/talknagish/react-native-turbo-starter
Going to open a discussion about it soon

**Reply by cortinico** [reply]
That's awesome. So we haven't yet defined a way to allow external contribs to the sample repos that @cipolleschi linked. Anyway, if someone is willing to spend some time into creating a new branch on their personal fork we could look into it and include it in the first party repos.

**Comment by satya164** [comment]
Hey @cipolleschi, thanks a lot for the guides. I was able to get this running successfully on iOS. However I have some issues on Android. In [this section](https://github.com/react-native-community/RNNewArchitectureLibraries/tree/feat/back-turbomodule#turbomodule-android-refactor-the-code-to-use-a-shared-implementation), it says:

> Open the example-library/android/src/main/java/com/rnnewarchitecturelibrary/CalculatorPackage.java file and replace all the instances of CalculatorModule with CalculatorModuleImpl

This doesn't seem right since `CalculatorModuleImpl` is not a native module, so returning `new CalculatorModule(reactContext)` in `getModule` won't work when using [the example `CalculatorPackage` here](https://github.com/react-native-community/RNNewArchitectureLibraries/tree/feat/back-turbomodule#turbomodule-android-convert-reactpackage-to-a-backward-compatible-turboreactpackage).

Edit: figured it out, since we specify 2 different sources, the appropriate `CalculatorModule` will be used for the package, so it doesn't require any changes. So this step in the guide is not correct.
